editor/ok_interface.py

```diff
@@ -93,116 +30,167 @@ def write(self, message):
+        actual = re.sub(r"Traceback.*\n\.\.\.\n(.*)", r"\1", actual)
+        return AreDifferent("\n".join(prompt), expected, actual)
+    else:
+        return Same("\n".join(prompt), result.strip())
+
+def process_case(case):
+    setup_success, setup_out = capture_output(case.console, case.setup.splitlines())
+    if not setup_success or "Traceback" in "".join(setup_out):
+        return FailureInSetup(setup_out)
+    interpret_success_overall = True
+    interpret_out_overall = []
+    for chunk in chunked_input(case.lines + case.teardown.splitlines()):
+        interpret_success, interpret_out = capture_output(case.console, chunk)
+        interpret_success_overall = interpret_success_overall and interpret_success
+        interpret_out_overall.append(process(interpret_out))
+    return FullTestCase(
+        interpret_success_overall,
+        interpret_out_overall)
 
-        elements.append((category, data))
-    return elements, error
-
-
-def create_test_case_from_block(block):
-    lines = block.strip().split("\n")
-
-    categorized_lines = list(categorize_test_lines(lines))
-    collapsed = collapse_test_lines(categorized_lines)
-    elements, error = process_test_errors(collapsed)
-
-    (_, [header]), *this_case = elements
-
-    match = re.match(r'(\S+) > Suite (\S+) > Case (\S+)', header)
-
-    problem = match.group(1).replace("-", " ").title()
-    suite = int(match.group(2))
-    case = int(match.group(3))
-    return TestCase(problem, suite, case, not error, this_case)
+def run_tests():
+    LOGGING_FORMAT = '%(levelname)s  | %(filename)s:%(lineno)d | %(message)s'
+    logging.basicConfig(format=LOGGING_FORMAT)
+    log = logging.getLogger('client')  # Get top-level logger
 
+    # noinspection PyUnresolvedReferences
+    from client.cli.ok import parse_input
+    from client.sources.ok_test.scheme import SchemeSuite
+    log.setLevel(logging.ERROR)
 
-def parse_test_data(raw_out):
-    blocks = raw_out.split("-" * 69)
-    blocks = blocks[1:-1]  # cut off the header + footer
-    cases = [create_test_case_from_block(block) for block in blocks]
+    args = parse_input(["--all", "--verbose", "--local"])
 
-    out = []
-    for case in cases:
-        if not out or out[-1]["problem"] != case.problem:
-            out.append({"problem": case.problem, "suites": [],
-                        "passed": all(x.passed for x in cases if x.problem == case.problem)})
-        if len(out[-1]["suites"]) != case.suite:
-            out[-1]["suites"].append([])
-        out[-1]["suites"][-1].append(case.export())
+    assign = assignment.load_assignment(None, args)
 
-    return out
+    result = []
+    for test in assign.specified_tests:
+        suites = []
+        for suite in test.suites:
+            assert isinstance(suite, SchemeSuite)
+            suites.append([process_case(case).dictionary for case in suite.cases])
+        result.append({
+            "problem": test.name,
```

Comment:
```suggestion
            "problem": test.name.replace("-", " ").title(),
```

To capitalize the problem names correctly.